chapter_1/2_kcenters.tex

```diff
@@ -67,3 +82,175 @@ \section{The \emph{k}-centers problem}
+\end{proof}
+
+\begin{remark}
+This is actually an algorithm for the symmetric problem under $2$-perturbation resilience and the asymmetric problem under $O(log^*k)$-perturbation resilience.
+\end{remark}
+
+Conversely, the asymmetric problem poses the issue that while $d(c(x),x)\leq r^*$, the symmetric distance may be arbitrarily large. To overcome this, we can define a set $A:=\{x|\forall y,d(y,x)\leq r^*\Rightarrow d(x,y)\leq r^*\}$.
+
+\begin{definition}
+The set $A$ \emph{respects the structure of the optimal clustering} if $C\subset A$ and for $x\in X\setminus A$, $A(x):=argmin_{a\in A}d(a,x)\in C\Rightarrow x\in C$.
+\end{definition}
+
+\begin{lemma}
+The two properties above hold for the asymmetric problem under 2-perturbation resilience.
+\end{lemma}
+
+\begin{proof}
+The sketch is given.\\
+
+The proof of the first statement was shown by Awasthi et al. for the symmetric problem. The proof of the asymmetric problem is the same.\\
+
+Suppose we have $c_i\in C_i$ and $q\in C$ such that $d(q,c_i)\leq r^*$. Then, expand all distances by 2 except for the distances from $q$ to $C_i$ which is expanded to be $2r^*$. By the lemma, the cost is $2r^*$ and since $q$ is $2r^*$ away from $C_i$, it replaces $c_i$ as the optimal center. This leads to a contradiction, as they may not exist in the same cluster. Therefore, both properties must hold for the asymmetric problem under 2-perturbation resilience.
+\end{proof}
+
+\subsection{$NP$-Hardness}
+
+Now let us consider the hardness of the problem under the $(2-\epsilon)$-perturbation stability. Note that this is a stronger approximation than that introduced by Balcan et al.
+
+\begin{definition}
+A clustering instance $(X,d)$ is $(\alpha,\epsilon)$-\emph{approximation stable} if for any partition $C'$ with cost $r'$, $r'\leq\alpha r^*\Rightarrow C'$ is $\epsilon$-close to the optimal cluster.
+\end{definition}
+
+\begin{remark}
+$(\alpha,\epsilon)$-perturbation stability implies $(\alpha,\epsilon)$-perturbation resilience. To briefly demonstrate this, the optimal cost for any $\alpha$-perturbation us at most $\alpha r^*$ under $d$, and if it violates this condition, it is $\epsilon$-far.
+\end{remark}
+
+\begin{theorem}
+The $k$-center problem under $(2-\epsilon)$-approximation stability is $NP$-hard, unless $NP=RP$.
+\end{theorem}
+
+\begin{proof}
+The sketch is given.\\
+
+Consider a reduction from the Perfect Dominating Set such that for all dominating sets of size at most $k$, every vertex is hit by one and only one denominator. The application of this construction to the $(2-\epsilon)$ problem is demonstrated by Ben-David and Reyzin. Ben-David and Reyzin showed that $\alpha$-perturbation resilience implies $x,c_i\in C_i$ and $c_j\in C_j$, $\alpha d(c_i,x)<d(c_j,x)$. Extend this by considering the YES case, where each denominator hits at least $\frac n{2k}$ vertices, the size of the clusters. Otherwise, there is at most 1 dominating set of size at most $k$.
```

Comment:
Could you add some more details here? A proof sketch is totally fine, but this one seems to be a bit quick. The last two sentences in particular ("Extend this by considering the YES case, where each denominator hits at least n 2k vertices, the size of the clusters. Otherwise, there is at most 1 dominating set of size at most k.") seem to skip a lot of steps and are difficult to follow.